enhancements/security/bmc-ca-certificate-support.md

```diff
@@ -0,0 +1,294 @@
+  securely using local-CA-signed certificates or self-signed certificates, so
+  that I can maintain a high level of security within the cluster.
+
+- As a cluster administrator, I want to update the CA certificates to address
+  various situations, such as renewing the validity of self-signed certificates
+  or trusting additional certificates issued by other local CAs, so that I can
+  ensure continued secure communication with BMCs.
+
+### Goals
+
+- Users provide CA certificates for IPI installation or node management after
+  IPI, and these certificates successfully validate the SSL connections with
+  baremetal BMCs.
+
+- Users update the CA certificates for their existing clusters, and the updated
+  certificates successfully validate the SSL connections with baremetal BMCs.
+
+### Non-Goals
+
+- Allow users to modify the CA certificate during the IPI installation process.
+
+- Provide an automatic CA certificate issuance feature for IPI installation or
+  cluster node management.
+
+- Support certificate validation for devices or components other than
+  baremetal BMCs.
+
+## Proposal
+
+- Modify [OpenStack Ironic][OpenStack Ironic] to add a new configuration option
+  named `default_verify_ca_path` in `ironic.conf` to allow providing a default
+  `verify_ca` path for all drivers. If `default_verify_ca_path` is set and
+  `driver_info["xxxx_verify_ca"]=true`, then use the value of
+  `default_verify_ca_path` as `verify_ca` instead of `true`. With this
+  approach, separate CA certificates can be specified for SSL connections with
+  BMCs without having to add them to the system's default certificate bundle.
+
+- Modify [ironic-image][Metal3 Ironic Container] to specify the CA certificates
+  in a pre-defined path to Ironic. The pre-defined path can be, e.g.
+  `/certs/ca/bmc`, which is the mount point for the CA certificates in
+  ironic-image. When starting the container, check if `/certs/ca/bmc` exists,
+  if it does, then:
+  - Write the CA certificates mounted in `/certs/ca/bmc` to a file, e.g.
+    `/tmp/bmc-tls.pem`.
+  - Launch a process to monitor changes to `/certs/ca/bmc`, if changes are
+    detected, then update `/tmp/bmc-tls.pem` accordingly.
+  - Set the `default_verify_ca_path` in `ironic.conf` using `/tmp/bmc-tls.pem`
+    as the value.
+
+- Modify [installer][OpenShift Installer] for control plane installation:
+  - Add a new optional field `platform.baremetal.bmcCACert` in
+    `install-config.yaml` to allow users to enter the contents of the CA
+    certificates.
+  - Perform a pre-validation of the SSL connection with the BMCs using the
+    provided CA certificates before the installation begins. If the validation
+    fails, an error message will be outputed to indicate which baremetal
+    failed. This helps to identify issues early on, especially for the
+    time-consuming IPI installation.
+  - Modify `startironic.sh.template`:
+    - Create the CA certificate files in `/tmp/bmc` of the bootstrap VM
```

Comment:
nit: this is really too specific. We don't need to agree on paths now, only on the design.